app/src/main/java/com/example/next_show/adapters/ShowAdapter.java

```diff
@@ -55,6 +60,22 @@ public void clear() {
 
     public void addAll(List<Show> list) {
         shows.addAll(list);
+        allShows = shows;
+        notifyDataSetChanged();
+    }
+
+    public void filter(String query){
+        Log.i("Adapter", "Filtering for " + query);
+        List<Show> queriedShows = new ArrayList<>();
+        for(Show s : shows){
+            if (s.getGenres().contains(query)) {
+                queriedShows.add(s);
+            }
+        }
+
+        // update the adapter and clear current items
+        shows.clear();
+        shows.addAll(queriedShows);
```

Comment:
Can you see why there might be a bug in these lines?

Reply:
Yes. I think because I clear the shows list directly (which holds all of the shows I grab from Trakt), I can't restore the previous state even if I remove all of the filters. I did create a variable above called `allShows` to perhaps hold all of the shows regardless of what the filter does however keep track of `allShows` and `shows` in the adapter was confusing.

Reply:
So, you iterate through the list of `shows` and then only include the shows whose genres match the query.  I think something that will help you figure out the bug is to consider: what shows will be included in the `shows` variable at the initial state, before and after filter 1, and then before and after filter 2 if you do all filters serially